Advanced/Sync Addressables/Assets/SyncAddressables/SyncBundleProvider.cs

```diff
@@ -23,10 +25,37 @@ internal void Start(ProvideHandle provideHandle)
                 Debug.LogError("the bundle failed " + provideHandle.Location.InternalId);
             provideHandle.Complete(this, m_AssetBundle != null, null);
         }
+
+        internal void Unload()
+        {
+            if (m_AssetBundle != null)
+            {
+                m_AssetBundle.Unload(true);
+                m_AssetBundle = null;
+            }
+        }
     }
     
     public override void Provide(ProvideHandle providerInterface)
     {
         new SyncAssetBundleResource().Start(providerInterface);
     }
+
+    public override void Release(IResourceLocation location, object asset)
+    {
+        if (location == null)
+            throw new ArgumentNullException("location");
+        if (asset == null)
+        {
+            Debug.LogWarningFormat("Releasing null asset bundle from location {0}.  This is an indication that the bundle failed to load.", location);
+            return;
+        }
+        var bundle = asset as SyncAssetBundleResource;
```

Comment:
Class `SyncAssetBundleResource` is an internal dummy which is instantiated in `SyncBundleProvider.Provide` only and is never saved to anywhere. So I am not sure if that cast will ever return `true`. You should probably cast to `AssetBundle` here.

Reply:
perhaps you were testing in "sync fast".  If you "use build data" or actually execute a real player, this will get hit.  And those are the only paths where an unload is actually valid.  So the above unload method will get called.